_core/page/head.php

```diff
@@ -12,142 +12,124 @@
-                $dat .= "[<a href='" . PHP_ASELF_ABS . "?mode=news' >Edit News/Boardlist</a>]";
-            }
-            $dat .= "[<a href='" . PHP_ASELF . "?mode=logout'>" . S_LOGOUT . "</a>]</div>";
+            $ribbon = ($admin) ? $this->adminRibbon() . "<hr>" : null;
+            
+            $dat .= '<body><div class="beforePostform" /><div id="boardNavDesktop">' . $this->get_cached_file(BOARDLIST) . '</div>
+                <div class="linkBar">[<a href="javascript:void(0);" id="settingsWindowLink">Settings</a>][<a href="' . HOME . '" target="_top">' . S_HOME . '</a>]</div><div class="boardBanner">' . $bannerImg . $boardTitle . '</div><hr>' . $ribbon . '<a id="top"></a>';
+            $dat .= (ENABLE_ADS) ? "<div class='ads aboveForm'>" . ADS_ABOVEFORM . '<hr></div></div>' : "</div>";
+        } else {
+            $dat .= "<body>";
         }
+
         return $dat;
-    } 
+    }
 
-	function headerJS($admin = false) {
-		//Going to serve as our ghetto mini-API for now
+    //Returns the HTML path to a random banner. $bannerAssets is stored in the board config.
+    private function randomBanner() {
+        //global $bannerAssets;
+        /*
+        $banner = ASSET_PATH . '/banners/' . BOARD_DIR . '/' . $bannerAssets[array_rand($bannerAssets)];
+  
+        return $banner;*/
+    }
+    
+    private function adminRibbon() {
+        $modes = array("panel", "reports", "appeals", "filters", "assets", "users", "settings", "rebuild", "logout");
+        $temp .= "<div class='adminRibbon'>";
+        $temp .= "[<a href='/" . BOARD_DIR . "/'>Return to Index</a>] ";
+        
+        foreach($modes as $mode) 
+            if (($mode === 'logout' || $mode === 'panel') || valid($mode)) $temp .= "[<a href='" . PHP_SELF_ABS . "?mode=admin&admin={$mode}'>" . ucfirst($mode) . "</a>] "; //Truly php has functions for everything
+            
+        $temp .= "</div>";
+        return $temp;
+    }
+    
+	private function headerJS($admin = false) {
+        global $cssArray;
 		$temp .= "var phpself = '" . PHP_SELF . "';";
-		$temp .= "var site = '//" . SITE_ROOT_BD . "';";
+		$temp .= "var site = '//" . SITE_ROOT . "';";
+        $temp .= "var cssPath = '" . CSS_PATH . "';";
 		$temp .= "var board = '" . BOARD_DIR . "';"; 
 		$temp .= "var jsPath = '" . JS_PATH . "';";
-		$temp .= (NSFW) ? 'var styleGroup = "nsfw";' : 'var styleGroup = "sfw";';
-		if (defined(EXTRA_SHIT)) $temp .= EXTRA_SHIT; 
+        $temp .= "var imgSRC = '" . PUBLIC_IMG_DIR . "';";
+        $temp .= "var thumbSRC = '" . PUBLIC_THUMB_DIR . "';";
+        //$temp .= "var inPanel = 'false';";
+		$temp .= (NSFW) ? 'var style_group = "nsfw";' : 'var style_group = "sfw";';
 
-		if ($admin) {
-			//Admin js variables go here
+		if ($admin) { //Admin js variables go here
+            $temp .= "var inPanel = 'true';";
 		}
 
 		$temp = "<script type='text/javascript'>" . $temp . "</script>";
 		
 		return $temp;
-
 	}
+    
+    private function get_cached_file($filename) {
```

Comment:
As `$my_log->update()` generates each index page, it's reading the board list file for each page iteration.

Example: For one post, to update a board with 5 index pages, saguaro reads the boardlist file 5 (each index) + 1 (thread page) = six times. This was to prevent unnecessary disk usage